### ISSUE TITLE
`enum_glob_use` is not triggered

### ISSUE THREAD

**Issue by mamekoro** [rust-lang/rust-clippy#11834]
### Summary

I thought the following reproducer would cause a compile error due to `#![deny(clippy::enum_glob_use)]`, but it compiled.

I'm new to clippy, am I missing something?

### Lint Name

enum_glob_use

### Reproducer

```rust
#![deny(clippy::enum_glob_use)]

pub enum Palette {
    Red,
    Green,
    Blue,
    Cyan,
    Magenta,
    Yellow,
}

impl Palette {
    pub fn to_rgb(self) -> u32 {
        use Palette::*;
        match self {
            Red => 0xFF0000,
            Green => 0x00FF00,
            Blue => 0x0000FF,
            Cyan => 0x00FFFF,
            // Fuchsia is the same color as Magenta.
            // Assume that the name Fuchsia was used in previous versions of the project
            // and has been replaced by the name Magenta.
            //
            // The current version does not have `Palette::Fuchsia`,
            // so this line `Fuchsia => 0xFF00FF` is a bug.
            //
            // I'd like to prevent this bug using `enum_glob_use`.
            Fuchsia => 0xFF00FF,
            Yellow => 0xFFFF00,
        }
    }
}

```

`cargo build` displays some warnings, but `enum_glob_use` does not appear to be working.

<details><summary>build log</summary>
<p>

```
warning: unreachable pattern
  --> src/lib.rs:29:13
   |
28 |             Fuchsia => 0xFF00FF,
   |             ------- matches any value
29 |             Yellow => 0xFFFF00,
   |             ^^^^^^ unreachable pattern
   |
   = note: `#[warn(unreachable_patterns)]` on by default

warning: unused variable: `Fuchsia`
  --> src/lib.rs:28:13
   |
28 |             Fuchsia => 0xFF00FF,
   |             ^^^^^^^ help: if this is intentional, prefix it with an underscore: `_Fuchsia`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: variable `Fuchsia` should have a snake case name
  --> src/lib.rs:28:13
   |
28 |             Fuchsia => 0xFF00FF,
   |             ^^^^^^^ help: convert the identifier to snake case (notice the capitalization): `fuchsia`
   |
   = note: `#[warn(non_snake_case)]` on by default

warning: `aaa` (lib) generated 3 warnings (run `cargo fix --lib -p aaa` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

</p>
</details>

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by Centri3** [comment]
This appears to trigger on the playground? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=958b71335ba82a1b14015665373b2dab

**Comment by mamekoro** [comment]
Ah, so `cargo build` does not automatically run clippy. My mistake.
